scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/Scheduler.java

```diff
@@ -977,9 +977,10 @@ String getTaskServerLogsByTag(String id, String taskTag) throws UnknownJobExcept
      * @throws NotConnectedException
      * @throws PermissionException
      */
-    Page<TaskState> getTaskStates(String taskTag, long from, long to, boolean mytasks,
-            boolean running, boolean pending, boolean finished, int offset, int limit)
-                    throws NotConnectedException, PermissionException;
+    Page<TaskState> getTaskStates(String taskTag, long from, long to,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this method to throw at most one checked exception instead of: org.ow2.proactive.scheduler.common.exception.NotConnectedException, org.ow2.proactive.scheduler.common.exception.PermissionException [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1160)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Method has 10 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00107)